lib/logstash/filters/geoip.rb

```diff
@@ -84,6 +84,16 @@ class LogStash::Filters::GeoIP < LogStash::Filters::Base
   # number of cache misses and waste memory.
   config :lru_cache_size, :validate => :number, :default => 1000
 
+  # If the ip field is an array: check which one is not a private ip. The first non private ip will be used.
+  config :filter_private_ips, :validate => :boolean, :required => false, :default => true
+
+  # String by which the ip field should be tokenized
+  config :ip_split_symbol, :validate => :string, :required => false, :default => ","
```

Comment:
The problem with the mutate split function is that it works inplace. If, for whatever reason, the user wants to keep the original value of the ip list, that would be impossible. Also I think it's more elegant to provide the functionality as a whole, to make it work standalone without requiring outside conditions. Makes it easier for the user and less error prone.